### DISCUSSION TITLE
Extracting grainId from grain plots

### DISCUSSION THREAD

**Discussion by SoeTho** [mtex-toolbox/mtex#1153] [Ask Anything] [answered]
Hello,

In one of the last versions you changed the information one gets by clicking on grains in grain maps. 

> figure;plot(grains)

Now grain properties are shown in the command window. 

> grainId = 151
phase = Forsterite
(x,y) = 496, 998
Euler = (244.8°,30.2°,92.5°)

This is useful and faster than before, so thank you very much! Since I often work with subsets where I need to manually delete certain grains, I was wondering if it would be possible to extract the grainId of the selected grain and save it directly or set the ebsd data of that grain to phase=0.

> ebsd2(grains(grains.id(151))).phase=0;

Thanks in advance for your help!
Best,
Sören

**Comment by ralfHielscher** [comment, marked as answer]
Hi Sören,

you can do
```matlab
plot(grains)
selectInteractive(grains)
```
Then you can select / unselect grains with the mouse. When ready do
```matlab
global indSelected
```
and the variable `indSelected` will contain the indices of the selected grains.

Ralf.

**Reply by SoeTho** [reply]
Works perfectly and saves me a lot of work. Thank you very much!
